haas_websocket/haas_rest_handler.py

```diff
@@ -0,0 +1,154 @@
+import os
+import json
+import requests
+import haas_websocket.haas_password_generator
+from dotenv import load_dotenv
+from google.cloud import secretmanager
+
+class TokenAuth():
+
+    def __init__(self):
+        load_dotenv()
+        self.username_key = os.getenv('HAAS_AUTH_USERNAME_KEY')
+        self.password_key = os.getenv('HAAS_AUTH_PASSWORD_KEY')
+        self.bearer_key = os.getenv('HAAS_BEARER_TOKEN_KEY')
+        self.refresh_key = os.getenv('HAAS_REFRESH_TOKEN_KEY')
+        self.api_endpoint = os.getenv('HAAS_API_ENDPOINT')
+        self.id = os.getenv('PROJECT_ID')
+        self.r_token = "initialized"
+
+    def secretManagerGet(self,secret_id):
+        if not self.id or not secret_id:
+            return None
+        try:
+            client = secretmanager.SecretManagerServiceClient()
+            parent = client.secret_path(self.id, secret_id)
+            sv_list = client.list_secret_versions
+            parent_list = sv_list(request={"parent": parent})
+            for version in parent_list:
+                if (version.state == 1):
+                    response = client.access_secret_version(request={"name": version.name})
+                    decoded = response.payload.data.decode("UTF-8")
+                    return decoded
+        except Exception:
+            return None
+
+    def secretManagerSet(self,secret_id,payload):
+        if not self.id or not secret_id or not payload:
+            return None
+        try:
+            client = secretmanager.SecretManagerServiceClient()
+            encoded_payload = payload.encode("UTF-8")
+            parent = f"projects/{self.id}/secrets/{secret_id}"
+            response = client.add_secret_version(
+                request={"parent": parent, "payload": {"data": encoded_payload}}
+            )
+            return response
+        except Exception:
+            print(f"(ERROR) Could not add a new secret version for the secret ID: {secret_id}")
+            return None
+
+
+    def signIn(self):
+        self.api_username = self.secretManagerGet(self.username_key)
+        self.api_password = self.secretManagerGet(self.password_key)
+
+        r = requests.post(self.api_endpoint + "oauth/token",
+                    json={"grant_type": "password",
+                    "username":self.api_username,
+                    "password":self.api_password})
+        json_response = json.loads(r.content)
+
+        self.b_token = json_response["access_token"]
```

Comment:
use .get to access values safely

Reply:
yeah, then the return will be None if the key doesn't exist. Only use when you have other error handling that will handle None though